sync_poltens.py

```diff
@@ -0,0 +1,342 @@
+        cl_00 = nmt.compute_full_master(f0, f0, binning)
+
+        cl = np.concatenate([binning.get_effective_ells()[None, :], cl_00, cl_22])
+        if is_Dell:
+            fname = "haslam_kband_spectra_dl.fits"
+        else:
+            fname = "haslam_kband_spectra_cl.fits"
+        hp.write_cl(str(DATA_DIR / fname), cl, overwrite=True)
+
+    iqu = hp.read_map(str(DATA_DIR / "transformed_iqu_fwhm2deg_corrected.fits"), field=(0, 1, 2))
+    for is_Dell in [True, False]:
+        binning = nmt.NmtBin(nside=nside, nlb=1, lmax=lmax, is_Dell=is_Dell)
+        f2 = nmt.NmtField(joint_mask, iqu[1:])
+        f0 = nmt.NmtField(joint_mask, [iqu[0]])
+
+        cl_22 = nmt.compute_full_master(f2, f2, binning)
+        cl_00 = nmt.compute_full_master(f0, f0, binning)
+
+        cl = np.concatenate([binning.get_effective_ells()[None, :], cl_00, cl_22])
+        if is_Dell:
+            fname = "frolov_2degfwhm_corr_dl.fits"
+        else:
+            fname = "frolov_2degfwhm_corr_cl.fits"
+        hp.write_cl(str(DATA_DIR / fname), cl, overwrite=True)
+
+    return 
+
+def PlotPowerspectra():
+    """ Plot the powerspectra of the Haslam and WMAP Kband maps, scaled
+    to 23 GHz, and plotted in uK^2.
+    """
+    dl = hp.read_cl(str(DATA_DIR / "haslam_kband_spectra_dl.fits"))
+    fig, ax = plt.subplots(1, 1)
+    ax.loglog(dl[0], dl[1], label=r"${\rm TT,~Haslam}$")    
+    ax.loglog(dl[0], dl[2], label=r"${\rm EE,~K~band}$")
+    ax.loglog(dl[0], dl[5], label=r"${\rm BB,~K~band}$")
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.set_xlabel(r"$\ell$")
+    ax.set_ylabel(r"$\ell (\ell + 1) / 2 \pi~C_\ell~({\rm \mu K}^2)$")
+    ax.legend(frameon=False)
+    ax.tick_params(direction="inout", which="both")
+    ax.set_title(r"${\rm Data~at~23~GHz}$")
+    fig.savefig("spectra_TTEEBB.pdf")
+
+    dl = hp.read_cl(str(DATA_DIR / "frolov_2degfwhm_corr_dl.fits"))
+    fig, ax = plt.subplots(1, 1)
+    ax.loglog(dl[0], dl[1], label=r"${\rm tt}$")    
+    ax.loglog(dl[0], dl[2], label=r"${\rm ee}$")
+    ax.loglog(dl[0], dl[5], label=r"${\rm bb}$")
+    ax.spines['top'].set_visible(False)
+    ax.spines['right'].set_visible(False)
+    ax.set_xlabel(r"$\ell$")
+    ax.set_ylabel(r"$\ell (\ell + 1) / 2 \pi~C_\ell~({\rm \mu K}^2)$")
+    ax.legend(frameon=False)
+    ax.tick_params(direction="inout", which="both")
+    ax.set_title(r"${\rm Frolov~transformed~spectra}$")
+    fig.savefig("frolov_spectra_tteebb.pdf")
+    return
+
+
+def PerformFitStokes(lmin=10, lmax=36):
+    """ Fit power law model to sycnhrotron spectrum over a determined
+    range of ells.
+    """
+    dl = hp.read_cl(str(DATA_DIR / "haslam_kband_spectra_dl.fits"))
+
+    fig, ax = plt.subplots(1, 1)
+    for (idx, label) in [(1, r"${\rm TT,~\gamma=}$"), (2, r"${\rm EE,~\gamma=}$"), (5, r"${\rm BB,~\gamma=}$")]:
+        pars, cov = curve_fit(PowerLaw, dl[0, lmin:lmax], dl[idx, lmin:lmax])
```

Comment:
also here, doesn't the fact that `dl` is a log change how you do the curve fitting?

Reply:
I haven't fitted a curve to any spectra of Frolov quantities yet, that function has only been applied to the combination of Haslam and WMAP.  

I thought the point of the transformation was that it is more appropriate to treat the (iqu) as Gaussian than the (IQU). Therefore, does it not make sense to fit the power law index in the log quantities?

Reply:
ok yes, you are right, we should fit power laws to the log quantities